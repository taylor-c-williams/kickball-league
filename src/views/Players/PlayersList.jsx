import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { getPlayers } from '../../services/players';

export default function PlayersList() {
	const [players, setPlayers] = useState([]);
	const [loading, setLoading] = useState(true);

	useEffect(() => {
		getPlayers()
			.then((res) => setPlayers(res))
			.finally(() => setLoading(false));
	}, []);

	if (loading) return <span className="loading">Loading...</span>;

	return (
		<div className="content">
			<h1>Players </h1>
			<ul>
				{players.map((player) => {
					return (
						<li key={player.id}>
							<Link to={`/players/${player.id}`} className="App-Link">
								{player.name}
							</Link>
						</li>
					);
				})}
			</ul>
			<Link to="/newPlayer">Add New Player</Link>
		</div>
	);
}
