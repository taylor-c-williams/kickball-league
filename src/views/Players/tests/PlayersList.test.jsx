import {
	waitForElementToBeRemoved,
	render,
	screen,
} from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import PlayersList from '../PlayersList';

it('renders PlayersList elements', async () => {
	const { container } = render(
		<Router>
			<PlayersList />
		</Router>
	);

	await waitForElementToBeRemoved(() => screen.queryByText(/Loading.../i));
	const textElement = await screen.findByText('Players', { exact: false });
	expect(textElement).toBeInTheDocument();
	expect(container).toMatchSnapshot();
});
