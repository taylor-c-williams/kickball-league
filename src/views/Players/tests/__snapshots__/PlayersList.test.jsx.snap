// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PlayersList elements 1`] = `
<div>
  <div
    class="content"
  >
    <h1>
      Players 
    </h1>
    <ul>
      <li>
        <a
          class="App-Link"
          href="/players/3"
        >
          player name
        </a>
      </li>
    </ul>
  </div>
</div>
`;
